src/client/components/UserList/UserListHeader/UserListHeader.tsx

```diff
@@ -19,7 +21,7 @@ const UserListHeader: React.FC<Props> = ({ isAdmin, roleNames }) => {
         <th className="user-list__header-cell">{t('userManagement.name')}</th>
         {!isAdmin && <th className="user-list__header-cell">{t('userManagement.role')}</th>}
         {isAdmin &&
-          roleNames.map((roleName: RoleName) => (
+          (Objects.isEmpty(filters.roles) ? roleNames : filters.roles).map((roleName: RoleName) => (
```

Comment:
create a hook, useRoleNames that will handle the filter there

Reply:
too complicated.

- that role list depends on the view, so it can't be placed into meta/users or meta/userRoles
- if we move the logic above into a hook we can use it to show the filtered list used to build the table columns but not the original one, which is used to show the filters

so this is the only logical way to avoid replicating that list.

Reply:
I dont understand the comment. dont we show the user roles only in the admin page ?

Reply:
yes, that's why at the moment there is a roleNames constant defined at view level, as suggested by you.

but that list it's used to display this
<img width="850" alt="Screenshot 2022-11-09 alle 12 24 26" src="https://user-images.githubusercontent.com/10047945/200818019-b2d98cfb-30bb-4bbc-ba01-d591b039d6ff.png">
and this
<img width="1397" alt="Screenshot 2022-11-09 alle 12 24 41" src="https://user-images.githubusercontent.com/10047945/200818046-a81164fb-5ed7-4e16-ab9c-9f2f51c44e77.png">

the first one is static and depends on the view.

the second one is dynamic and depends on the state (ui.userManagement.filters).

Reply:
Can't you have 2 hooks ? UseRoleNames and useFilteredRileNames ?

Reply:
ok